addons/html_editor/static/tests/list/automatic_list_detection.test.js

```diff
@@ -143,6 +143,28 @@ test("should convert a bullet list into a numbered list", async () => {
     expect(getContent(el)).toBe(`<ol><li placeholder="List" class="o-we-hint">[]<br></li></ol>`);
 });
 
+test("typing '[] ' should create checklist", async () => {
+    const { el, editor } = await setupEditor("<p>[]</p>");
+    insertText(editor, "[] ");
+    expect(getContent(el)).toBe(
+        `<ul class="o_checklist"><li placeholder="List" class="o-we-hint">[]<br></li></ul>`
+    );
```

Comment:
Can you add one more step `HISTORY_UNDO` and add a check if its back to normal (`[] `).